File: app/src/main/java/com/example/bhagavadgita/ViewModel/MainViewModel.java
package com.example.bhagavadgita.ViewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.example.bhagavadgita.Models.Chapters;

import java.util.List;

public class MainViewModel extends ViewModel {

    private final MutableLiveData<List<Chapters>> chaptersLiveData = new MutableLiveData<>();

    public MainViewModel() {

    }

    public LiveData<List<Chapters>> getChaptersLiveData() {
        return chaptersLiveData;
    }

    public void setChapters(List<Chapters> chapters) {
        chaptersLiveData.setValue(chapters);
    }




}

File: app/src/main/java/com/example/bhagavadgita/ui/main/viewModel/MainViewModel.kt
package com.example.bhagavadgita.ui.main.viewModel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.bhagavadgita.data.model.Chapters
import com.example.bhagavadgita.data.repository.MainRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject


@HiltViewModel
class MainViewModel @Inject constructor(
    private val mainRepository: MainRepository,
    ) : ViewModel() {

    private val _chapters = MutableLiveData<List<Chapters>>()
    val chapters: LiveData<List<Chapters>>
        get() = _chapters

    init {
        fetchData()
    }


    private fun fetchData() {
        viewModelScope.launch {

            mainRepository.getAllChapters()
            mainRepository.chapters.observeForever { data ->
                _chapters.postValue(data)
            }
        }
    }

}
